mirrord/operator/src/client.rs

```diff
@@ -628,39 +548,39 @@ impl OperatorApi<PreparedClientCert> {
     {
         self.check_feature_support(config)?;
 
-        // Check if the deployment is empty if so then
-        let is_empty_deployment = match &config.target.path {
-            Some(Target::Deployment(target)) => {
-                let deployment_api = get_k8s_resource_api::<Deployment>(
-                    &self.client,
-                    config.target.namespace.as_deref(),
-                );
-
-                let deployment = deployment_api
-                    .get(&target.deployment)
-                    .await
-                    .map_err(|error| OperatorApiError::KubeError {
-                        error,
-                        operation: OperatorOperation::FindingTarget,
-                    })?;
-
-                !deployment
-                    .status
-                    .map(|status| status.available_replicas > Some(0))
-                    .unwrap_or_default()
-            }
-            _ => false,
-        };
+        if target.containers_count() > 1
+            && config
+                .target
+                .path
+                .as_ref()
+                .is_some_and(|target| target.container().is_none())
+        {
+            progress.warning(&format!(
+                "Target has multiple containers, but no container was specified by user!\
+                    mirrord will pick a random container `{}`.",
+                target.container().unwrap_or_default()
```

Comment:
No point in including this, because inside this `if`, `target.container()` always returns `None`, doesn't it? and then we'd always get the default `""` here.